feat: implementation of SPARQUE navigation

skoch-intershop opened the pull request:
## PR Type
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no API changes)
[ ] Build-related changes
[ ] CI-related changes
[ ] Documentation content changes
[ ] Application / infrastructure changes
[ ] Other: <!--Please describe.-->

## What Is the Current Behavior?

Main navigation (catalog pages, family pages and product detail pages) collect necessary data from the icm/solr.

Issue Number:

## What Is the New Behavior?

Main navigation collect necessary data from the SPARQUE wrapper API.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

[AB#108984](https://dev.azure.com/intershop-com/cefd1005-00a7-4c79-927f-a16947d1b2e6/_workitems/edit/108984)

[AB#105629](https://dev.azure.com/intershop-com/Products/_workitems/edit/105629)

[AB#108938](https://dev.azure.com/intershop-com/Products/_workitems/edit/108938)

skoch-intershop committed:
feat: implementation of sparque navigation (#108983)

SGrueber commented on src/app/core/services/sparque-categories/sparque-categories.service.spec.ts:
```diff
@@ -0,0 +1,66 @@
+    sparqueCategoriesService = TestBed.inject(SparqueCategoriesService);
+  });
+
+  it('should always delegate to sparque api service when called', () => {
```
can you add a describe for testing the getTopLevelCategories ? - somehow it is not clear by the test descriptions

SGrueber commented on src/app/core/models/sparque-category-tree/sparque-category-tree.interface.ts:
```diff
@@ -0,0 +1,11 @@
+import { SparqueCategory } from 'ish-core/models/sparque-category/sparque-category.interface';
+
+export interface SparqueCategoryTreeResponse {
```
the interface should always match the file pattern name - if it is an interface (and not a model) it is clear it is a response - so the name of this interface should be SparqueCategoryTree

skoch-intershop committed:
feat: implementation of sparque browsing (#108983)

skoch-intershop committed:
feat: usage of sparque wrapper api v3 (#108983)

skoch-intershop committed:
feat: integration of sparque products usage (#108983)

skoch-intershop commented on src/app/core/store/shopping/products/products.effects.ts:
```diff
@@ -186,46 +203,45 @@ export class ProductsEffects {
+                      ? // handle Sparque filter
+                        loadFilterSuccess({
+                          filterNavigation: {
+                            filter,
```
hierarchy logic will resolved by mapper

skoch-intershop committed:
chore: category facet optimisation (#108983)

andreassteinmann committed:
chore: update files to match develop

andreassteinmann committed:
fix: enable breadcrumbs for Sparque categories

- Modify breadcrumb selector to work with categories that have categoryPath
  even if not completely loaded (needed for Sparque)
- Simplify Sparque category service to return empty tree for individual
  category requests since Sparque API doesn't support this
- Remove debug logging from category effects

andreassteinmann committed:
fix: enable product detail pages with Sparque config

- Refactor SparqueProductMapper to properly set completeness levels
- fromData uses level 3 for detail pages, fromListDataSingle uses level 2 for lists
- Eliminates code duplication while maintaining dynamic level behavior

andreassteinmann committed:
fix: enable default sorting on product lists with Sparque config

- Don't send sorting parameter when sortKey is 'default' in SparqueProductsService
- Matches SOLR behavior where default sorting means no explicit sorting parameter
- Fixes issue where selecting 'Default sorting' showed no products with Sparque

andreassteinmann committed:
refactor: improve service providers and shopping store effects

- Enhance products service provider with robust error handling and complete interface
- Improve suggestions service provider with pure reactive patterns and better encapsulation
- Update filter selectors for better Sparque integration
- Refine products effects and tests for improved functionality
- Optimize search effects and tests for better performance
- Update shopping store tests to reflect changes

skoch-intershop committed:
chore: refactoring after rebasing

andreassteinmann committed:
fix: remove unused imports and mocks

andreassteinmann committed:
fix: display product description for Sparque products

- Use shortDescription as fallback for longDescription in SparqueProductMapper
- Update test expectations to reflect the fallback behavior
- Ensures Sparque products show description in product detail view
